src/NuGet.Licenses/Controllers/LicenseController.cs

```diff
@@ -149,8 +149,64 @@ private ActionResult DisplayLicenseFromFile(string identifier, bool isException)
         private ActionResult DisplayCompositeLicenseExpression(LicenseOperator licenseExpressionRoot, string licenseExpression)
         {
             var meaningfulSegments = _licenseExpressionSegmentator.GetLicenseExpressionSegments(licenseExpressionRoot);
+
             var allSegments = _licenseExpressionSegmentator.SplitFullExpression(licenseExpression, meaningfulSegments);
 
+            var stopwatch = Stopwatch.StartNew();
+            foreach (var segment in allSegments)
+            {
+                if (segment.Type == CompositeLicenseExpressionSegmentType.LicenseIdentifier || segment.Type == CompositeLicenseExpressionSegmentType.ExceptionIdentifier)
```

Comment:
I would invert the `if` statement to decrease statement, or even "move" it to `foreach`:
```
foreach (var segment in allSegments.Where(s => s.Type == CompositeLicenseExpressionSegmentType.LicenseIdentifier || s.Type == CompositeLicenseExpressionSegmentType.ExceptionIdentifier))
```
(maybe making a function `IsLicenseOrException(segment) => segment.Type == CompositeLicenseExpressionSegmentType.LicenseIdentifier || segment.Type == CompositeLicenseExpressionSegmentType.ExceptionIdentifier`) would make the whole thing a bit more readable.